state-chain/runtime/src/lib.rs

```diff
@@ -1191,6 +1191,26 @@ impl_runtime_apis! {
 			}
 		}
 
+		fn cf_min_egress_amount(asset: Asset) -> AssetAmount {
+			use pallet_cf_ingress_egress::MinimumEgress;
+			use cf_chains::assets::{eth, dot, btc};
+
+			match ForeignChain::from(asset) {
+				ForeignChain::Ethereum => MinimumEgress::<Runtime, EthereumInstance>::get(
```

Comment:
A demo of redundancy of having [a map instead of a single-value](https://github.com/chainflip-io/chainflip-backend/pull/4458#discussion_r1469626108):

For each `$CHAIN` in all chains:
- get the storage for `$CHAIN`
- in the latter storage fetch the value for the Asset of `$CHAIN`